app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt

```diff
@@ -50,13 +53,28 @@ abstract class TwoPaneFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         setupSlidingPane()
+        observeFolderName()
         observeThreadEvents()
     }
 
     private fun setupSlidingPane() {
         slidingPaneLayout.lockMode = SlidingPaneLayout.LOCK_MODE_LOCKED
     }
 
+    private fun observeFolderName() = with(mainViewModel) {
+
+        Utils.waitInitMediator(currentFolder, isInSearch).observe(viewLifecycleOwner) { (folder, isInSearch) ->
+
+            val displayedFolder = (if (isInSearch) {
+                Folder().apply { name = getString(R.string.searchFolderName) }
+            } else {
+                folder
+            }) ?: return@observe
+
+            rightPaneFolderName.value = getString(R.string.noConversationSelected, displayedFolder.getLocalizedName(context))
```

Comment:
What if we create a function inside TwoPaneFragment like `onFolderNameChanged` or `updateSecondPaneFolderName` or whatever, and we call it here and the parent overrides it